File: algorithms_python/sort/insertion_rec.py
#source: https://www.geeksforgeeks.org/recursive-insertion-sort/
def insertionRecSort(arr, n):
	if n <= 1:
		return

	insertionRecSort(arr, n-1)
	last = arr[n-1]
	j = n - 2

	while (j>=0 and arr[j] > last):
		arr[j+1] = arr[j]
		j = j -1

	arr[j+1] = last

File: algorithms_java/sort/InsertionRec.java
/*Source: https://www.geeksforgeeks.org/recursive-insertion-sort/ */
public class InsertionRec{
	static void insertionRecSort(Long arr[], int n){
		if(n <= 1)
			return;

		insertionRecSort(arr, n - 1);

		Long last = new Long(arr[n-1]);
		int j = n - 2;

		while(j>= 0 && arr[j] > last)
		{
			arr[j+1] = arr[j];
			j--;
		}
		arr[j+1] = last;
	}
}
